src/hangar/records/queries.py

```diff
@@ -110,10 +110,11 @@ def _traverse_dataset_data_records(self, dataset_name):
             with datatxn.cursor() as cursor:
                 # with datatxn.cursor() as cursor:
                 dataRecordsExist = cursor.set_range(startDatasetRecordCountRangeKey)
+                dataRecKeySubString = f'{startDatasetRecordCountRangeKey.decode()}:'.encode()
```

Comment:
Instead of explicitly specifying the "colon" `:` as the seperator, we have defined the "seperator" constant in the config file here:

https://github.com/tensorwerk/hangar-py/blob/dc08edb25c66ea9b2d029ac8ba8c771de2b18bbb/src/hangar/config_hangar.yml#L14-L15

At the top of the file, define:

```python
SEP = config.get('hangar.seps.key')
```

like we do in the `parsing.py` file here:

https://github.com/tensorwerk/hangar-py/blob/dc08edb25c66ea9b2d029ac8ba8c771de2b18bbb/src/hangar/records/parsing.py#L21-L26

Which would change the addition to:

```python
 dataRecKeySubString = f'{startDatasetRecordCountRangeKey.decode()}{SEP}'.encode()
```

The better solution (but one which may require some debuging work) would probably be to add the `SEP` to the end of the `dset data count` key which is incred/decred every time a sample is added/removed. 

This would be in the code here (and the reverse code as well)

https://github.com/tensorwerk/hangar-py/blob/dc08edb25c66ea9b2d029ac8ba8c771de2b18bbb/src/hangar/records/parsing.py#L286-L289

Remember to look at the `repo._details()` call output to see this for yourself. 

![image](https://user-images.githubusercontent.com/17350115/57858977-0bfbfa80-77c0-11e9-9f47-d320181f1294.png)

However, the base solution does work with that modification!

Reply:
Let me make the tweak and add test cases as soon as I could make some time. Speaking of dset count key, a completely unrelated issue, is there any reason we are looping over the whole dataset samples when we do `len(dataset)`. We could just fetch this key and return the count?

Reply:
I had previously had some issues with the count being inacurate in certain scenarious and that method of counting was a quick hack so i could get the rest of the system working... however those issues have long been fixed, and this is just a remnant of that old hack.

 If you want to streamline that, it would be a welcome improvement!